src/Diagnostics/FxCop/System.Runtime.Analyzers/Core/Design/DefineAccessorsForAttributeArguments.Fixer.cs

```diff
@@ -103,10 +108,19 @@ public class DefineAccessorsForAttributeArgumentsFixer : CodeFixProvider
             return symbolEditor.GetChangedDocuments().First();
         }
 
-        private async Task<Document> MakePublic(Document document, SyntaxNode getMethod, CancellationToken cancellationToken)
+        private async Task<Document> MakePublic(Document document, SyntaxNode getMethod, SyntaxNode property, CancellationToken cancellationToken)
         {
+            // Clear the accessibility on the getter.
             DocumentEditor editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
-            editor.SetAccessibility(getMethod, Accessibility.Public);
+            editor.SetAccessibility(getMethod, Accessibility.NotApplicable);
+
+            // If the containing property is not public, make it so
+            var propertyAccessibility = editor.Generator.GetAccessibility(property);
+            if (propertyAccessibility != Accessibility.Public)
+            {
+                editor.SetAccessibility(property, Accessibility.Public);
+            }
```

Comment:
Shouldn't we also set the setter's accessibility to the previous accessibility of the property (in case where setter doesn't have accessibility explicitly specified) to ensure that we don't make it more accessible than it was.